docs/examples/hypervisors/kvm/modules/kvm-hypervisor/scripts/configure-secondary-vnics.sh

```diff
@@ -0,0 +1,181 @@
+#!/bin/bash -eu
+
+# Configure a secondary VNIC on a bare metal instance
+# Prerequisite: Add a secondary VNIC to the instance using Console/SDK/API.
+
+# An error exit function
+function error_exit {
+	echo "$1" 1>&2
+	exit 1
+}
+
+function usage {
+cat <<EOF
+Usage: $0 [-n namespace] [-h]
+    This script configures a secondary VNIC on a bare metal instance.
+    It shows the list of available secondary VNICs and configures the one you select.
+    [-n namespace] - Optional parameter to specify the namespace in which the VNIC needs to be configured.
+                     It is configured in the default namespace if this option is not specified.
+    [-h] - Shows the usage information
+EOF
+
+}
+
+#Initialize NS to empty string. Secondary VNIC will be configured in the default namespace by default.
+NS=
+
+# Parse command line arguments
+while getopts ":hn:" opt; do
+    case $opt in
+        h)
+            usage
+            exit 0
+            ;;
+        n)
+            NS=$OPTARG
+            ;;
+        \?)
+            echo "Invalid option: -$OPTARG" >&2
+            usage
+            exit 1
+            ;;
+        :)
+            echo "Option -$OPTARG requires an argument." >&2
+            exit 1
+            ;;
+    esac
+done
+
+# Make sure the script is run as root
+[[ $EUID -eq 0 ]] || error_exit "This script must be run as root"
+
+# Load 8021q module for vlan tagging support: https://en.wikipedia.org/wiki/IEEE_802.1Q
+modprobe 8021q || error_exit "Failed to load 8021q module"
+
+IMDSURL="http://169.254.169.254/opc/v1/vnics/"
+
+# Get list of private IPs for all VNICs configured on the instance, except the first one in the list.
+# Note: the first VNIC is the primary VNIC, which is already configured on the instance
+# Note: jq is not available by default on Oracle Linux images. Using Python to avoid dependency on Internet connectivity.
+options=($(curl -s $IMDSURL | python -c 'import sys, json; print "\n".join([vnic["privateIp"] for vnic in json.load(sys.stdin)])' | tail -n +2))
+
+# Sanity check that $options isn't empty
+! [ -z $options ] || error_exit "Unable to list the private IPs from $IMDSURL"
+
+# Show the list of private IPs and let the user select the index of the IP to be configured on the instance.
+
+COUNTER=1
+while [ $COUNTER -lt $((${#options[@]} + 1 )) ]; do
```

Comment:
Please format the content with the new while loop. Easier to understand and debug if necessary.